components/video-player.tsx

```diff
@@ -109,6 +110,11 @@ const VideoPlayer = forwardRef<HTMLVideoElementWithPlyr, Props>(({ playbackId, p
           'This is an old browser that does not support MSE https://developer.mozilla.org/en-US/docs/Web/API/Media_Source_Extensions_API',
         );
       }
+
+      if (currentTime) {
+        video.currentTime = currentTime;
+      }
```

Comment:
I'm getting a little bit of inconsistent behavior. I think there is a race condition here because:

- in the parent component the `router.query.time` value starts off undefined and then gets populated asynchronously
- I believe VideoPlayer is loading first with `currentTime` undefined and then the `currentTime` prop gets updated

We probably want to extract this to it's own useEffect, that depends on the `currentTime` prop:

```js
useEffect(() => {
  const video = videoRef.current;
  if (currentTime && video) {
    video.currentTime = currentTime;
  }
}, [currentTime, videoRef])
```

Are you seeing some inconsistent behavior when you QA it? Or is it just me?

Reply:
Well it shouldn't ever be set to `undefined` due to the guards in the parent element, but I haven't been able to reproduce the inconsistent behavior. Regardless, though, you're right that this should definitely be its own effect so that the behavior matches the `currentTime` prop you'd expect from a video element!